Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/VaccinatedPerson.kt

```diff
@@ -50,9 +50,14 @@ data class VaccinatedPerson(
             .maxByOrNull { it.vaccinatedAt }
             ?: return null
 
-        val immunityAt = newestFullDose.vaccinatedAt.toDateTimeAtStartOfDay().plus(IMMUNITY_WAITING_PERIOD)
+        val immunityAt = newestFullDose.vaccinatedAt
```

Comment:
Wouldn't it be simpler to just use the local date and get Days.daysBetween() as I did for days since installation? 
eg. val daysSinceInstallation: Int
        get() = Days.daysBetween(dayOfInstallation, today).days

Reply:
with val today: LocalDate
        get() = Instant.now().toLocalDateUserTz()

Reply:
Both should work.
I'm not sure if I find `daysBetween` simpler, just different?

Using `daysBetween` would also mean loosing resolution, currently we get back a `Duration` and ever consumer of the API can use a time resolution of their choice.

If this is not a blocker for you, I'd prefer to not change this.

Reply:
A duration does not give you precise information about the how many date changes it contains. We are counting date changes here, not 24h intervals, I assume. The information needed for this is lost once converted to a duration.
I believe this solution is error prone.

Reply:
I think counting date changes is a UI detail, business logic wise, a `Duration` until immunity is reached (a countdown) makes sense to me. But I have no strong feelings about this if you want to change this.

Reply:
@chiljamgossow is right here. Counting 24h intervals will never be completely right. At the start/end of daylight saving time a day is 23h/25h long.

So I think @d4rken's proposal is correct, that actually days should be counted.

Reply:
using Days.daysBetween now